packages/core/src/search/match.ts

```diff
@@ -253,6 +256,36 @@ function matchesDateValue(resourceValue: string, operator: Operator, filterValue
   }
 }
 
+/**
+ * Builds the column value for a date/time parameter.
+ * Tries to parse the date string.
+ * Silently ignores failure.
+ * @param value - The FHIRPath result.
+ * @returns The date/time string if parsed; undefined otherwise.
+ */
+function buildDateTimeColumn(value: unknown): string | undefined {
+  if (!value) {
+    return undefined;
+  }
+  if (typeof value === 'string') {
+    try {
+      const date = new Date(value);
+      return date.toISOString();
```

Comment:
The tests that are failing are because we're now comparing date.toISOString(), e.g. `1990-01-01T00:00:00.000Z` to the raw filter string, e.g. `1990-01-01`. It seems like this would work out okay if we also pass `filterValue` through `buildDateTimeColumn`?

Reply:
Yeah...

I started down the path of doing a more proper "search IR" idea that we've talked about, but that broke even more tests.

For now, I'm going to just use our "search param details" to differentiate between "date" and "datetime".